CONTRIBUTING.md

```diff
@@ -0,0 +1,504 @@
+- [Debugging](#debugging)
+  - [Running Cocoapods Locally](#running-cocoapods-locally)
+- [Pull Requests](#pull-requests)
+  - [Pull Request Checklist](#pull-request-checklist)
+  - [Step 1: Open Issue](#step-1-open-issue)
+  - [Step 2: Design (optional)](#step-2-design-optional)
+  - [Step 3: Fork The Repo](#step-3-fork-the-repo)
+  - [Step 4: Work your Magic](#step-3-work-your-magic)
+  - [Step 5: Commit](#step-4-commit)
+  - [Step 6: Pull Request](#step-5-pull-request)
+  - [Step 7: Merge](#step-6-merge)
+- [Troubleshooting](#troubleshooting)
+- [Related Repositories](#related-repositories)
+- [Finding Contributions](#finding-contributions-to-work-on)
+- [Code of Conduct](#code-of-conduct)
+- [Security Issue Notifications](#security-issue-notifications)
+- [Licensing](#licensing)
+
+## Getting Started
+
+To get started with contributing to the AWS iOS SDK, first make sure you have the latest version of [Xcode](https://developer.apple.com/xcode/) installed as well as [cocoapods](https://guides.cocoapods.org/using/getting-started.html). You can install cocoapods by simply running:
+
+```bash
+sudo gem install cocoapods
+```
+
+Then make sure you clone the project by running:
+
+```bash
+git clone https://github.com/aws-amplify/aws-sdk-ios
+```
+
+The project itself is broken up into 2 Xcode projects so it's important to understand what you want to change and open the correct project. If you open the project at the root of the repo, that contains most of the individual SDK's, but if you are looking to make a change to our Auth SDK's (including `AWSMobileClient`), head to the folder called `/AWSAuthSDK` and there is another Xcode project in there you can open. Wherever you start, whether its at the root project or in the AWSAuthSDK's project, make sure you run `pod install` and open the workspace. Example of commands to open the main SDK project that lives at the root of the repo  below:
+
+```bash
+cd aws-sdk-ios
+pod install
+xed .
+```
+
+Conversely if you wanted to make a change to our Auth SDK's the initial workflow might be:
+
+```bash
+cd aws-sdk-ios
+cd AWSAuthSDK
+pod install
+xed .
+```
+
+## Tools
+
+[Xcode](https://developer.apple.com/xcode/) and [cocoapods](https://guides.cocoapods.org/using/getting-started.html) are used for all build and dependency management.
+
+Some other widely used dependencies are:
+
+[Mantle](https://github.com/Mantle/Mantle) is what we use for deserialization and serialization and any modeling with JSON in our AWS iOS SDK.
+
+[SocketRocket](https://github.com/facebook/SocketRocket) is used for web socket connections.
+
+### Integration Testing Setup
+
+For our integration tests, which run on your development machine we use XCTest and an aws configuration file to set up to set up credentials. To set it up to build those integration tests on your machine, you will need to create a `credentials.json` file in inside AWSCoreTests> Resources>credentials.json and it should have the following format:
```

Comment:
Consider the wording:

Our integration tests are designed to execute on your development machine and talk to live AWS services. `we should probably have something here about best practices of not talking to their own production services??`  These tests are based on XCTest and require a json file with your credentials.   You will need to create a file named `credentials.json` located in ..... and it should have the following format:

Reply:
done